File: Cpp/Example5-1-RasterBase/Example5-1-RasterBase.cpp
// Example1RasterBase.cpp : 此文件包含 "main" 函数。程序执行将在此处开始并结束。
//

#include <gdal_priv.h>

#include <iostream>

using namespace std;

int main() {
  GDALAllRegister();

  string srcFile = getenv("GISBasic");
  srcFile = srcFile + "/../Data/Raster/berry_ali_2011127_crop_geo.tif";

  GDALDataset* img = (GDALDataset*)GDALOpen(srcFile.c_str(), GA_ReadOnly);
  if (!img) {
    return 1;
  }

  int imgWidth = img->GetRasterXSize();   //图像宽度
  int imgHeight = img->GetRasterYSize();  //图像高度
  int bandNum = img->GetRasterCount();    //波段数
  int depth = GDALGetDataTypeSize(img->GetRasterBand(1)->GetRasterDataType()) /
              8;  //图像深度

  cout << "宽度：" << imgWidth << '\n';
  cout << "高度：" << imgHeight << '\n';
  cout << "波段数：" << bandNum << '\n';
  cout << "深度：" << depth << '\n';

  GDALClose(img);
  img = nullptr;
}

File: Python/example-5-1-raster-base.py
from osgeo import gdal
import os

# 注册所有支持的格式（在 Python 中通常自动完成，但显式调用也无妨）
gdal.AllRegister()

# 获取环境变量 GISBasic 并构造文件路径
gis_basic = os.getenv("GISBasicRepo")
if not gis_basic:
    raise EnvironmentError("环境变量 GISBasicRepo 未设置")

src_file = os.path.join(gis_basic, "Data", "Raster", "berry_ali_2011127_crop_geo.tif")

# 打开栅格数据集
dataset = gdal.Open(src_file, gdal.GA_ReadOnly)
if dataset is None:
    print("无法打开文件:", src_file)
    exit(1)

# 获取图像信息
img_width = dataset.RasterXSize      # 图像宽度
img_height = dataset.RasterYSize     # 图像高度
band_num = dataset.RasterCount       # 波段数

# 获取第一个波段的数据类型，并计算每个像素的字节数（深度）
band = dataset.GetRasterBand(1)
data_type = band.DataType
depth = gdal.GetDataTypeSize(data_type) // 8  # 转换为字节

# 输出信息
print(f"宽度：{img_width}")
print(f"高度：{img_height}")
print(f"波段数：{band_num}")
print(f"深度：{depth}")

# 关闭数据集（Python 中通常靠引用计数自动释放，但显式置 None 更安全）
dataset = None